plugins/modules/npm.py

```diff
@@ -300,10 +301,14 @@ def main():
         no_bin_links=dict(default=False, type='bool'),
         force=dict(default=False, type='bool'),
     )
-    arg_spec['global'] = dict(default=False, type='bool')
+    arg_spec['global'] = dict(type='bool')
     module = AnsibleModule(
         argument_spec=arg_spec,
-        required_if=[('state', 'absent', ['name'])],
+        required_if=[
+            ('state', 'absent', ['name']),
+            ("global", False, ["path"]),
+        ],
+        required_one_of=[('path', 'global')],
```

Comment:
```suggestion
```
I'd remove this, since it's not really needed:
1. If none is specified, Ansible will already say that `path` is required (since `global` defaults to `false`).
2. If `global` is specified and explicitly set to `false`, this won't have an effect anyway (and the `required_if` will enforce that `path` is specified).
3. And if `global` is specified and set to `true`, everything is as before (`path` can be specified or not, and is ignored).

Reply:
Yeah, on ~~second~~ third thought, I am overcomplicating it.